clang/include/clang/CConv/ConstraintVariables.h

```diff
@@ -229,8 +229,10 @@ class PointerVariableConstraint : public ConstraintVariable {
   void insertQualType(uint32_t TypeIdx, QualType &QTy);
   // This function tries to emit an array size for the variable.
   // and returns true if the variable is an array and a size is emitted.
-  bool emitArraySize(std::ostringstream &Pss, uint32_t TypeIdx, bool &EmitName,
-                     bool &EmittedCheckedAnnotation, bool Nt);
+  bool emitArraySize(std::stack<std::string> &CheckedArrs, uint32_t TypeIdx,
+                     bool &EmitName, bool &AllArray, bool &ArrayRun, bool Nt);
+  void addArrayAnnotations(std::stack<std::string> &CheckedArrs,
+                           std::deque<std::string> &EndStrs);
```

Comment:
How to think about the above two functions? Their names are similar one is emitting the array size and other is adding an annotation, which probably includes the size.

Reply:
Added comments clarifying